python/shotgun_menus/entity_field_menu.py

```diff
@@ -221,7 +234,12 @@ def entity_filter(et):
                     bubble_fields[field] = {
                         "name": display_name,
                         "valid_types": entity_types,
-                        "valid_type_names": [shotgun_globals.get_type_display_name(et) for et in entity_types],
+                        "valid_type_names": [
+                            shotgun_globals.get_type_display_name(
+                                et,
+                                project_id=self._project_id,
+                            ) for et in entity_types
+                        ],
```

Comment:
the `except Exception` below feels a bit brutal? would it make sense to reduce the scope, either in terms of lines of code or in terms of faults to trap? Not saying we have to, just pointing it out :)